src/ApacheOrcDotNet.OptimizedReader/_Constants.cs

```diff
@@ -8,6 +9,8 @@ public static class Constants
         public static DateTime UnixEpochUtc = new(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
         public static DateTime UnixEpochUnspecified = new(1970, 1, 1, 0, 0, 0, DateTimeKind.Unspecified);
 
+        public static int NumPossibleStreams = Enum.GetValues<StreamKind>().Length;
```

Comment:
This assumes `StreamKind` values will always range from 0-X.  You should use `Max` here.  Also I'd probably just use a `Dictionary`.  This isn't inner-loop stuff.

Reply:
I've updated the buffer length, but kept the array since the comparison will be called several times while filling the buffers and the dictionary lookup/comparison is a bit slower that using the array directly.

Reply:
We did shave some nanoseconds off our 1.2s latency from AWS.  :p  This is fine.  It's a pattern we use often.